llvm/lib/Target/AMDGPU/GCNSchedStrategy.cpp

```diff
@@ -1524,9 +1582,6 @@ bool PreRARematStage::sinkTriviallyRematInsts(const GCNSubtarget &ST,
     MachineInstr *MI = Entry.first;
     MachineInstr *OldMI = Entry.second;
 
-    // Remove OldMI from BBLiveInMap since we are sinking it from its MBB.
-    DAG.BBLiveInMap.erase(OldMI);
```

Comment:
I'm not sure why you removed this line and added DAG.BBLiveInMap = DAG.getBBLiveInMap()  below

Reply:
Based on the way I'm reading the code and the comment, it seems the intention of this line is to remove any live-in registers that are defined by OldMI, as we are rematerailizing the registers. But, what this line actually does is remove the entire LiveIn LiveRegSet if OldMI is a key in the map.

It seems to me that we need to update the set of live-ins for the impacted region to reflect the eliminated cross-BB live ranges. Rather than doing this per-register, I've just recalculated the entire map after all the changes to LIS were made.

That said, perhaps this should be addressed separately?

Reply:
That's a good idea to address that in a separate PR.

Reply:
Okay, I've removed it from this PR.